src/common/s3.rs

```diff
@@ -2,18 +2,20 @@ use aws_sdk_s3::config;
 
 use crate::aws::ClientBuilder;
 
-pub(crate) struct S3ClientBuilder;
+pub(crate) struct S3ClientBuilder {
+    pub force_path_style: Option<bool>,
+}
 
 impl ClientBuilder for S3ClientBuilder {
     type Client = aws_sdk_s3::client::Client;
 
-    fn build(config: &aws_types::SdkConfig) -> Self::Client {
-        let config = config::Builder::from(config).build();
-        aws_sdk_s3::client::Client::from_conf(config)
-    }
+    fn build(&self, config: &aws_types::SdkConfig) -> Self::Client {
+        let mut builder = config::Builder::from(config);
+
+        if let Some(true) = self.force_path_style {
+            builder = builder.force_path_style(true);
+        }
```

Comment:
I think you could avoid the `if let Some` if you just do: `builder = builder.set_force_path_style(self.force_path_style)`. The `set_*` variants on the AWS SDK take `Option`s.

Reply:
Done! I opted to use `force_path_style` so I could unwrap to `true` by default in case someone forgets to set the value in `S3ClientBuilder`. That way it maintains vector's previous behavior even in a programming error.